clustering/cluster.go

```diff
@@ -15,6 +15,7 @@ type Pointer interface {
 type Cluster struct {
 	Centroid *geo.Point
 	Pointers []Pointer
+	Children []*Cluster // can be used to trace down the clustering tree.
```

Comment:
As we talked about, it could be clearer to call these 'parents', because the two clusters make a single cluster (as used).